palace/utils/geodata.hpp

```diff
@@ -51,10 +52,59 @@ void AttrToMarker(int max_attr, const mfem::Array<int> &attrs, mfem::Array<int>
 void AttrToMarker(int max_attr, const std::vector<int> &attrs, mfem::Array<int> &marker);
 
 // Helper function to construct the bounding box for all elements with the given attribute.
-void GetBoundingBox(mfem::ParMesh &mesh, int attr, bool bdr, mfem::Vector &min,
-                    mfem::Vector &max);
-void GetBoundingBox(mfem::ParMesh &mesh, const mfem::Array<int> &marker, bool bdr,
-                    mfem::Vector &min, mfem::Vector &max);
+void GetAxisAlignedBoundingBox(mfem::ParMesh &mesh, int attr, bool bdr, mfem::Vector &min,
+                               mfem::Vector &max);
+void GetAxisAlignedBoundingBox(mfem::ParMesh &mesh, const mfem::Array<int> &marker,
+                               bool bdr, mfem::Vector &min, mfem::Vector &max);
+
+// Struct describing a bounding box in terms of the center and face normals. The
+// normals specify the direction from the center of the box.
+struct BoundingBox
+{
+  // The central point of the bounding box.
+  std::array<double, 3> center;
+  // Vectors from center to the midpoint of each face.
+  std::array<std::array<double, 3>, 3> normals;
+  // Whether or not this bounding box is two dimensional.
+  bool planar;
+  // Compute the area of the bounding box spanned by the first two normals.
+  double Area() const;
+  // Compute the volume of a 3D bounding box. Returns zero if planar.
+  double Volume() const;
+  // Compute the lengths of each axis.
+  std::array<double, 3> Lengths() const;
+  // Compute the deviation in degrees of a vector from each of the normal
+  // directions.
+  std::array<double, 3> Deviation(const std::array<double, 3> &direction) const;
```

Comment:
Non inline methods for these calculations allows `lumpedelement.hpp` to be cleaner and avoid having `#include <Eigen/Dense>`